backend/src/middleware/validation/index.js

```diff
@@ -22,10 +22,27 @@ const validateUrl = async (resolve, root, args, context, info) => {
   }
 }
 
+const validateComment = async (resolve, root, args, context, info) => {
+  const COMMENT_MIN_LENGTH = 1
+  const content = args.content.replace(/<(?:.|\n)*?>/gm, '').trim()
+  if (!args.content || content.length < COMMENT_MIN_LENGTH) {
+    throw new UserInputError(`Comment must be at least ${COMMENT_MIN_LENGTH} character long!`)
+  }
+  const NO_POST_ERR_MESSAGE = 'Comment cannot be created without a post!'
+  const { postId } = args
+  if (!postId) {
+    throw new UserInputError(NO_POST_ERR_MESSAGE)
+  }
+  /* eslint-disable-next-line no-return-await */
+  return await resolve(root, args, context, info)
```

Comment:
> [<img alt="ALau2088" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ALau2088) **Authored by [ALau2088](https://github.com/ALau2088)**
_<time datetime="2019-05-29T00:07:35Z" title="Wednesday, May 29th 2019, 2:07:35 am +02:00">May 29, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -22,10 +22,29 @@ const validateUrl = async (resolve, root, args, context, info) => {
   }
 }
 
+const validateComment = async (resolve, root, args, context, info) => {
+  const COMMENT_MIN_LENGTH = 1
+  const content = args.content.replace(/<(?:.|\n)*?>/gm, '').trim()
+  if (!args.content || content.length < COMMENT_MIN_LENGTH) {
+    throw new UserInputError(
+      `Comment must be at least ${COMMENT_MIN_LENGTH} character long!`
+    )
+  }
+  const NO_POST_ERR_MESSAGE = 'Comment cannot be created without a post!'
+  const { postId } = args
+  if (!postId) {
+    throw new UserInputError(NO_POST_ERR_MESSAGE)
+  }
+
+  return await resolve(root, args, context, info)
```


@mattwr18 I am getting an ES-Linting error on line 39. error  Redundant use of `await` on a return value  no-return-await. Did you encounter this before? Any suggestions for a workaround?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-29T11:13:02Z" title="Wednesday, May 29th 2019, 1:13:02 pm +02:00">May 29, 2019</time>_

---


maybe you need to save the returned value from the `await resolve(root, args, context, info)` to a variable and return that?

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-29T19:30:06Z" title="Wednesday, May 29th 2019, 9:30:06 pm +02:00">May 29, 2019</time>_

---


@ALau2088  `return` automatically `awaits` therefore its redundant.
The Solution would be to `return resolve(...)` I guess

Reply:
> [<img alt="ALau2088" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ALau2088) **Authored by [ALau2088](https://github.com/ALau2088)**
_<time datetime="2019-05-30T22:43:12Z" title="Friday, May 31st 2019, 12:43:12 am +02:00">May 31, 2019</time>_

---


@mattwr18 @ulfgebhardt Thanks for the input guys!

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-31T19:48:05Z" title="Friday, May 31st 2019, 9:48:05 pm +02:00">May 31, 2019</time>_

---


@ALau2088 it seems that `return await resolve` is in fact different from `return resolve`. I guess, 99.9% of all cases it should make no difference and if it does, it's probably a code smell.

Nevertheless, I experiences issues when I removed the `await` [here](https://github.com/Human-Connection/Human-Connection/blob/master/backend/src/middleware/validation/index.js#L7). So if you *want* disable the linter, you can do so by adding a comment (see the linked LOC above :point_up: ).